nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/main/java/org/apache/nifi/processors/grpc/ListenGRPC.java

```diff
@@ -153,70 +162,83 @@
+                .maxInboundMessageSize(maxMessageSize);
 
-        if (useSecure && sslContext != null) {
+        if (useSecure) {
             // construct key manager
             if (StringUtils.isBlank(sslContextService.getKeyStoreFile())) {
                 throw new IllegalStateException("SSL is enabled, but no keystore has been configured. You must configure a keystore.");
             }
 
-            final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm(),
-                    sslContext.getProvider());
+            final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
             final KeyStore keyStore = KeyStore.getInstance(sslContextService.getKeyStoreType());
             try (final InputStream is = new FileInputStream(sslContextService.getKeyStoreFile())) {
                 keyStore.load(is, sslContextService.getKeyStorePassword().toCharArray());
             }
             keyManagerFactory.init(keyStore, sslContextService.getKeyStorePassword().toCharArray());
 
-            SslContextBuilder sslContextBuilder = SslContextBuilder.forServer(keyManagerFactory);
+            final SslContextBuilder sslContextBuilder = SslContextBuilder.forServer(keyManagerFactory);
 
             // if the trust store is configured, then client auth is required.
             if (StringUtils.isNotBlank(sslContextService.getTrustStoreFile())) {
-                final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm(),
-                        sslContext.getProvider());
+                final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
                 final KeyStore trustStore = KeyStore.getInstance(sslContextService.getTrustStoreType());
                 try (final InputStream is = new FileInputStream(sslContextService.getTrustStoreFile())) {
                     trustStore.load(is, sslContextService.getTrustStorePassword().toCharArray());
                 }
                 trustManagerFactory.init(trustStore);
-                sslContextBuilder = sslContextBuilder.trustManager(trustManagerFactory);
-                sslContextBuilder = sslContextBuilder.clientAuth(io.netty.handler.ssl.ClientAuth.REQUIRE);
+                sslContextBuilder.trustManager(trustManagerFactory);
+                sslContextBuilder.clientAuth(ClientAuth.REQUIRE);
             } else {
-                sslContextBuilder = sslContextBuilder.clientAuth(io.netty.handler.ssl.ClientAuth.NONE);
+                sslContextBuilder.clientAuth(ClientAuth.NONE);
             }
-            sslContextBuilder = GrpcSslContexts.configure(sslContextBuilder);
-            serverBuilder = serverBuilder.sslContext(sslContextBuilder.build());
+            GrpcSslContexts.configure(sslContextBuilder);
```

Comment:
This line can be removed since `GrpcSslContexts.configure()` is a helper method to return a new `SslContextBuilder`, which is not used.

Reply:
Not sure. `GrpcSslContexts.configure()` returns the `SslContextBuilder` instance passed in (like a builder).
The interesting part that it also configures some things on `SslContextBuilder`. That's why I don't think it can be removed.

Reply:
You are correct, after looking more closely at `GrpcSslContexts.configure()`, it does appear to be setting some protocol options based on the runtime platform.